### PR TITLE
Fix handling of phi nodes in LowerConstExprs.

### PR DESCRIPTION
**Pull request by zuban32**
Put newly created insertElement instructions at the ends
of the incoming blocks instead of putting them immediately before
the phi thus breaking the module.

### CODE REVIEW

File: lib/SPIRV/SPIRVLowerConstExpr.cpp [hunk 1]
```diff
@@ -162,14 +162,16 @@ void SPIRVLowerConstExpr::visit(Module *M) {
                          [LowerOp](Value *V) { return LowerOp(V); });
           Value *Repl = nullptr;
           unsigned Idx = 0;
+          auto *PhiII = dyn_cast<PHINode>(II);
+          auto *InsPoint = PhiII ? &PhiII->getIncomingBlock(OI)->back() : II;
```
Comment (AlexeySachkov) [inline lib/SPIRV/SPIRVLowerConstExpr.cpp:166]: What about using [`getFirstInsertionPt`](https://llvm.org/doxygen/classllvm_1_1BasicBlock.html#a161fd4e9fa5367f64c2a4c9e921c3ad3)?

File: test/constexpr_phi.ll [hunk 2]
```diff
@@ -0,0 +1,56 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc -spirv-text --spirv-ext=+SPV_INTEL_function_pointers -o %t.spt
+; RUN: FileCheck < %t.spt %s --check-prefix=CHECK-SPIRV
+; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_INTEL_function_pointers -o %t.spv
```
Comment (AlexeySachkov) [inline test/constexpr_phi.ll:4]:
```suggestion
; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_INTEL_function_pointers -o %t.spv
; RUN: llvm-spirv %t.spv -to-text -o %t.spt
; RUN: FileCheck < %t.spt %s --check-prefix=CHECK-SPIRV

```

File: lib/SPIRV/SPIRVLowerConstExpr.cpp (hunk 1 above)
Comment (zuban32) [reply lib/SPIRV/SPIRVLowerConstExpr.cpp:166]: AFAIR getFirstInsertionPt return the first instruction in the BB we can insert new inst before, but we can't do that here as phi should use that instructions

File: test/constexpr_phi.ll (hunk 2 above)
Comment (zuban32) [reply test/constexpr_phi.ll]: Fixed

File: lib/SPIRV/SPIRVLowerConstExpr.cpp (hunk 1 above)
Comment (AlexeySachkov) [reply lib/SPIRV/SPIRVLowerConstExpr.cpp:166]: > AFAIR getFirstInsertionPt returns the first instruction in the BB we can insert new inst before, but we can't do that here as phi already uses that instructions

Ah, got it - we need to insert new instruction into a predecessor BB, so it is already known/computed prior encountering phi - right?

**Comment by AlexeySachkov** [comment]
@zuban32, could you please rebase your PR one more time to incorporate build fixes in it?

**Pull request merged**